Library/Homebrew/completions.rb

```diff
@@ -296,7 +296,15 @@ def generate_fish_subcommand_completion(command)
           next unless FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING.key? type
 
           named_arg_function = FISH_NAMED_ARGS_COMPLETION_FUNCTION_MAPPING[type]
-          named_args << "__fish_brew_complete_arg '#{command}' -a '(#{named_arg_function})'"
+          named_arg_prefix = "complete -f -c brew -n '__fish_brew_command #{command}; and not __fish_seen_argument"
+
+          named_args << if type == :formula && named_args_types.include?(:cask)
```

Comment:
```suggestion
          named_args << if type == :formula && named_args_types.include?(:cask) && command_options(command).key?("--cask")
```

Some commands, like `cleanup`, accept both formula and cask arguments but don't provide the `--formula` or `--cask` options. In these cases, I think I'd prefer to use the existing logic. Tbh, though, these commands should probably have those options added.

Reply:
So will it be better If we just check `command_options(command).key?("--formula") && command_options(command).key?("--cask")`? By the way thanks for the `command_options` tips since I have little knowledge of Homebrew's codebase.

Reply:
Yeah, that makes sense. And that should take care of the `{installed,outdated}_{formula,cask}` issue too, I think.